xbmc/pvr/recordings/PVRRecordings.cpp

```diff
@@ -76,6 +76,33 @@ void CPVRRecordings::Update()
   CServiceBroker::GetPVRManager().PublishEvent(PVREvent::RecordingsInvalidated);
 }
 
+void CPVRRecordings::UpdateInProgressSize()
+{
+  CSingleLock lock(m_critSection);
```

Comment:
The implementation looks a bit weird with all the locks and unlocks. Why is m_recordings not locked while iterating it?

Reply:
I literally copied the `Update()` function above this one and modified it. You're right, this should not leaving the lock at all.